Update transition map details. (#6)
flatland-association/flatland-book: getting-started/env.md

- [**Observations:**](#observations) what can each agent "see"?
- [**Rewards:**](#rewards) what is the metric used to evaluate the agents?

🗺️ Environment
---

Flatland is a 2D rectangular grid environment of arbitrary width and height, where the most primitive unit is a cell. Each cell has the capacity to hold a
single agent (train).

An agent in a cell can have a discrete orientation direction which represents the cardinal direction the agent is pointing to. An agent can move to a subset of
adjacent cells. The subset of adjacent cells that an agent is allowed to transition to is defined by a 4-bit transition map representing possible transitions in
4 different directions.

![basic_railway_elements.drawio.png](../assets/images/basic_railway_elements.drawio.png)

> *10 basic cells modulo rotation enable us to implement any realworld railway network in the flatland env*

Reviewer: ```suggestion
*10 basic cells modulo rotation enable us to implement any real-world railway network in the flatland env*
```
real-world or real world ?
see:
https://www.merriam-webster.com/dictionary/real-world
or
https://en.wiktionary.org/wiki/real_world

This gives a set of 30 valid transitions in total (see `#` giving number of rotations).

Agents can only travel in the direction they are currently facing. Hence, the permitted transitions for any given agent depend both on its position and on its
direction. Transition maps define the railway network in the flatland world. One can implement any real world railway network within the Flatland environment by

> manupulating the transition maps of cells.

Reviewer: ```suggestion
manipulating the transition maps of cells.
```

> For more information on transtion maps checkout [environment information](../environment/environment_information)!

Reviewer: ```suggestion
For more information on transition maps checkout [environment information](../environment/environment_information)!
```

↔️ Actions
---

The trains in Flatland have strongly limited movements, as you would expect from a railway simulation. This means that only a few actions are valid in most
cases.

Here are the possible actions:

- **`DO_NOTHING`**:  If the agent is already moving, it continues moving. If it is stopped, it stays stopped. Special case: if the agent is at a dead-end, this
  action will result in the train turning around.
- **`MOVE_LEFT`**: This action is only valid at cells where the agent can change direction towards the left. If chosen, the left transition and a rotation of
  the agent orientation to the left is executed. If the agent is stopped, this action will cause it to start moving in any cell where forward or left is
  allowed!

> - **`MOVE_FORWARD`**: The agent will move forward. This action will start the agent when stopped. At switches, this will chose the forward direction.

Reviewer: ```suggestion
- **`MOVE_FORWARD`**: The agent will move forward. This action will start the agent when stopped. At switches, this will choose the forward direction.
```
https://www.merriam-webster.com/dictionary/choose

> - **`MOVE_RIGHT`**: The same as deviate left but for right turns.

Reviewer: _deviating_ instead of _deviate_ ?

Reviewer 2: as turning - instead of deviate : deviate ~ differs to smth.

The provided observations are defined in [envs/observations.py](https://gitlab.aicrowd.com/flatland/flatland/blob/master/flatland/envs/observations.py)
```

> Each of the provided observation has its strengths and weaknesses. However, it is unlikely that you will be able to solve the problem by using any single one of

Reviewer: ```suggestion
Each of the provided observations has its strengths and weaknesses. However, it is unlikely that you will be able to solve the problem by using any single one of
```

In **Flat**land 3, rewards are only provided at the end of an episode by default making it a sparse reward setting.

The episodes finish when all the trains have reached their target, or when the maximum number of time steps is reached.

The actual reward structure has the following cases:

> - **Train has arrived at it's target**: The agent will be given a reward of 0 for arriving on time or before the expected time. For arriving at the target later

Reviewer: ```suggestion
- **Train has arrived at its target**: The agent will be given a reward of 0 for arriving on time or before the expected time. For arriving at the target later
```
see:https://www.merriam-webster.com/dictionary/it's#did-you-know
